### ISSUE TITLE
Serious issue around "vendor-supported solutions"

### ISSUE THREAD

**Issue by NoahKunin** [ombegov/a130#11]
Can OMB clarify what they mean by:

>All IT systems and services operate only vendor-supported solutions...

https://github.com/ombegov/a130/blame/gh-pages/pages/policy.md#L75

I think I probably know what you're going for, insofar as you don't want agencies implementing technology that is not longer supported. But clearly OMB recognizes the untold numbers of open-source technologies fundamental to both government and industry that are supported by open source communities (either in addition to our without vendor support, which also may or may not include industry support both in terms of $ and technical expertise/contributions)

Git itself, of course, is a good example of this. :wink: 

Beyond open source community support, other technologies (whether open sourced or not) are supported by IT professionals already on staff in the civil service in many different government agencies.

Thoughts @rebeccawilliams @justgrimes before I PR?

**Comment by theplaz** [comment]
Perhaps instead mention having a plan to make sure updates are made?

**Comment by dbirch-usgs** [comment]
I agree. Including language that could limit organizations to pay for services is a bad idea. Considering free and open source options provides a means for cost savings.

**Comment by NoahKunin** [comment]
Put another way: it's completely appropriate that A-130 demand that we only use "supported technologies" and that agencies plan for, and even cite where that support is going to come from. But A-130 should be neutral on the origin of that support.

**Comment by dsmorgan77** [comment]
To put a finer point on @NoahKunin's rewording, the language is inconsistent with January 2011 memorandum on Technology Neutrality (https://www.whitehouse.gov/sites/default/files/omb/assets/egov_docs/memotociostechnologyneutrality.pdf). 

Recommend rewording to have the agency ensure that "All IT systems and services operate technology that meets the particular operational needs of the agency by considering factors such as performance, cost, security, interoperability, ability to share or reuse, and availability of quality support. Agency planning and budgeting shall activities incorporate migration planning and resourcing to accomplish this requirement."

**Comment by bnordgren** [comment]
I think this stresses the importance of limiting the above restrictions to Enterprise IT and having separate policy for Research IT (#16). Requiring "vendor supported" solutions cripples those who depend on the freedom to select the most appropriate Linux platforms to run cutting edge open source software maintained (or not) by small groups of scientists. Vendors don't support anything until it is well accepted, but by then, it's not really science any more. We're not just being difficult, developmental prototypes are our reason for being, and we're heavy users of unique IT assets.

I think @dsmorgan77 is closer to the mark (good call on the technology neutrality), but there is a painful tendency to place "All" IT in one bucket, then put the responsibility for planning and budgeting at the agency level. Whenever I read things like that, I can just see the next generation of needless hurdles being set up because someone forgot about us _again_. :) (editorial comment for the above: "Agency planning and budgeting shall ~~activities~~ incorporate...")

**Comment by cew821** [comment]
The use of open source software is definitely not constrained to "Research IT" (though @bnordgren's comments in #16 are definitely worth a read to see the challenges he faces in a research context).

For example, most of the software that powers the modern web is open source. The most successful user-facing web products today are built primarily using open source software frameworks, databases and languages. Use of proprietary software stacks to deliver web applications is not the norm.

Each team should use the best tools that make the most sense for the product they are attempting to implement ([see play 8 of the Digital Services Playbook](https://playbook.cio.gov/#play8)). In many cases, and for web applications especially, this will most likely heavily involve the use of open source software.

@dsmorgan77's language looks good.

**Comment by DanielSchuman** [comment]
I commented on this elsewhere, and am reproducing here. Doesn't add much to the convo, but as Noah pointed out, useful have all comments in one place.

In section 5 (policy) b (governance) 2 (CIO defined development process), part e requires All IT systems and services operate only vendor-supported solutions, and planning and budgeting activities incorporate ...."

What is meant by "vendor-supported solutions?" Does it mean outside vendors online, or vendors inside government? If government is capable of building and supporting solutions, then this requirement may be inapt or need to be combined.

Parsing the language, does it mean:
(1) if a solution comes from a vendor, then that vendor must support it? or
(2) if a solution comes from a vendor, then a vendor (not necessarily the same one) must support it? or
(3) all solutions must come from vendors, and a vendor (that vendor?) must support it

**Comment by kmclean** [comment]
"Vendor-supported" could easily be interpreted to pre-emptively exclude Open Source altogether by someone wishing to do so. While that interpretation is rather pedestrian, a COR (Contracting Officer's Representative) / COTR (Contracting Officer's Technical Representative) could use such reasoning as a check-in-the-box as they evaluate options. 

While it may seem a stretch, it has happened before--for years in DoD many non-technical CORs/COTRs unknowingly categorized "F/OSS" (which was encouraged) as "Freeware" (which was banned) because of similarly ambiguous/unclear language. In 2009 a memo had to be issued clarifying the definition.

As mentioned in #21, it would be preferable to see F/OSS explicitly addressed.

**Comment by benjaminellison** [comment]
I have provided some suggested alterations to this text in a pull request (#69, I believe).

**Comment by kmclean** [comment]
*Relative Comments from #67*

The FOSS option to CPIC section (84eafe7) looks very good. Thanks!

In Update policy.md (929912c), the use of "unsupported, legacy systems" narrows the widow of ambiguity, but doesn't close it completely. Since the intent is clearly to deal with legacy systems, perhaps "legacy systems which are no longer supported" would close the window. Additionally, it treats legacy proprietary and abandoned F/OSS projects equally.

**Comment by thecapacity** [comment]
I'm not so sure "updates are made" or is a good measure of whether or not something should stay around... I doubt anyone's made updates to fax machines recently, but I think we'd agree it's ok for the government to use them.

I agree with the intent being a good one, but I also think there is some "leeway" for "if it ain't broke..." other wise we might force agencies into expensive "end of life" contracts as a means to keep perfectly good systems running within the compliance framework.
